hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/shell/TestTextCommand.java

```diff
@@ -67,39 +75,194 @@ public void testDisplayForAvroFiles() throws Exception {
     assertEquals(expectedOutput, output);
   }
 
+  @Test
```

Comment:
We might want to Test a scenario where the stream is read in very small chunks (e.g., 1 or 2 bytes at a time) in the multi-byte read() method. This can confirm no edge cases remain where you might loop incorrectly or miss a final newline.

Reply:
Good idea! I added this test, for both Avro and sequence file. It works by setting `io.file.buffer.size` to `2` before the read. (Default used by other tests is `4096`.)

Reply:
Thanks